core/market/decentralized/src/negotiation/requestor.rs

```diff
@@ -87,6 +93,58 @@ impl RequestorBroker {
         Ok(())
     }
 
+    pub async fn counter_proposal(
+        &self,
+        subscription_id: &SubscriptionId,
+        prev_proposal_id: &str,
+        proposal: &ClientProposal,
+    ) -> Result<String, ProposalError> {
+        // TODO: Everything should happen under transaction.
+        // TODO: Check if subscription is active
+        // TODO: Check if this proposal wasn't already countered.
+        let prev_proposal = self
+            .db
+            .as_dao::<ProposalDao>()
+            .get_proposal(prev_proposal_id)
+            .await
+            .map_err(|e| {
+                ProposalError::FailedGetProposal(prev_proposal_id.to_string(), e.to_string())
+            })?
+            .ok_or_else(|| {
+                ProposalError::ProposalNotFound(
+                    prev_proposal_id.to_string(),
+                    subscription_id.clone(),
+                )
+            })?;
+
+        if &prev_proposal.negotiation.subscription_id != subscription_id {
+            Err(ProposalError::ProposalNotFound(
```

Comment:
I'm not sure what are we actually revealing here? Because the same error will be returned in case Proposal just wasn't found. So how anyone can distinguish?

Reply:
but if we were returning not authorized this would reveal that proposal of given id exists

Reply:
Aaa, so there's no problem with this code? This is just note for future, that we shouldn't return not authorized here?